src/main/java/org/jabref/logic/importer/IdBasedParserFetcher.java

```diff
@@ -61,8 +63,9 @@ default void doPostCleanup(BibEntry entry) {
             return Optional.empty();
         }
 
-        try (InputStream stream = new BufferedInputStream(getURLForID(identifier).openStream())) {
-            List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
+        try {
+            HttpResponse<String> response = Unirest.get(getURLForID(identifier).toString()).asString();
```

Comment:
Well, this now has the disdvantage that the Stream does not get closed. That's why try(....) is preferable.

Reply:
Ah sorry, got confused by the diff.
Why is this ByteArrayInputStream needed? Is there no way to directly get the response with unirest?

Reply:
Caused by the interface

Reply:
Not sure whats better but we should stick to one: unirest vs our UrlDownloader

Reply:
We could directly use jsoup. We use that in dfiferent cases

Reply:
Jsoup is for a DOM that needs to be parsed. I don't need that here. I just need the body contents.